File: twilio_client/twilio_client.py
from twilio.rest import Client
import twilio


class TwilioClient():
    def __init__(self, config):
        self.config = config
        # if 'API_KEYS' in config:
        account_sid = config['API_KEYS']['TWILIO_ACCOUNT_SID']
        auth_token = config['API_KEYS']['TWILIO_AUTH_TOKEN']
        self.client = Client(account_sid, auth_token)
        # if 'TWILIO' in config:
        self.from_number = config['TWILIO']['FROM_PHONE_NUMBER']
        self.message = config['TWILIO']['SMS_MESSAGE']

    def sendSMS(self, phone_number):
        ''' Tries to send a message to the user, does not send if client has
        requested to STOP receiving messages'''
        try:
            message = self.client.messages.create(
                to=phone_number,
                from_=self.from_number,
                body=self.message)
            print(message.sid)
        except twilio.base.exceptions.TwilioRestException:
            print("Message not sent, user said to STOP")

File: app/twilio_client/twilio_client.rb
require 'twilio-ruby'
require 'logger'

class TwilioClient
    def initialize(config)
        puts config.groups
        @config = config
        @logger = Logger.new(STDOUT)
        account_sid = config['API_KEYS']['TWILIO_ACCOUNT_SID']
        auth_token = config['API_KEYS']['TWILIO_AUTH_TOKEN']
        @client = Twilio::REST::Client.new(account_sid, auth_token)
        @from_number = config['TWILIO']['FROM_PHONE_NUMBER']
        @message = config['TWILIO']['SMS_MESSAGE']
    end

    def sendSMS(phone_number)
        begin
        @client.messages.create(
            to: phone_number,
            from: @from_number,
            body: @message
        )
        rescue
            @logger.warn("Message not sent, user said to STOP")
        end
    end

end